src/pyhf/typing.py

```diff
@@ -0,0 +1,132 @@
+import os
```

Comment:
I still need to actually review this, but if this is only providing the types can this get moved out to a separate PR that we merge in first so that things stay as tight as possible for scope? Some of our PRs have gotten big enough that it becomes difficult to understand later what the atomic scope of things was (e.g. the staterror problem we have now).

Reply:
tricky because there are bugs caught in this PR that are fixed related to resolving that would not have otherwise been caught without the typehints added. This is all being imported in.

Reply:
#1931, #1933 for the particular bugs